nucleus/admin/config-api/src/main/java/org/glassfish/config/support/TranslatedConfigView.java

```diff
@@ -149,13 +144,12 @@ private static Object getTranslatedValue(Object value) {
                 Config config = configResolver().getConfig();
                 ConfigValue configValue = config.getConfigValue(matchValue);
                 String newValue = configValue.getValue();
-                if (newValue != null && !newValue.isEmpty()) {
-                    stringValue = m3.replaceFirst(Matcher.quoteReplacement(m3.group(1) + newValue + m3.group(3)));
+                if (newValue != null && !newValue.isEmpty() || defaultValue != null) {
+                    stringValue = m3.replaceFirst(Matcher.quoteReplacement(m3.group(1) + (newValue == null ? defaultValue : newValue) + m3.group(3)));
```

Comment:
Actually I think you're missing an empty check.
You can get into this block with a non-null empty `newValue` if `defaultValue` is set.

`newValue` = `""`
`defaultValue` = `"asd"`

`if (newValue != null && !newValue.isEmpty() || defaultValue != null)` evaluates to _true_ due to `defaultValue`
`stringValue = m3.replaceFirst(Matcher.quoteReplacement(m3.group(1) + (newValue == null ? defaultValue : newValue) + m3.group(3)));` evaulates to `""` since `newValue` is not null